### PR TITLE
Bugfix/484405 fix downtime subflow

### PR DESCRIPTION
**Pull request by Guilhermealvestake**
fix downtime subflow

### CODE REVIEW

File: src/Take.Blip.Builder/FlowManager.cs [hunk 1]
```diff
@@ -331,7 +331,8 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
 
                         await ProcessGlobalOutputActionsAsync(context, flow, lazyInput, inputTrace, state, linkedCts.Token);
 
-                        await _inputMessageHandlerAggregator.OnFlowProcessedAsync(state, message, _applicationNode, linkedCts.Token);
+                        await _inputMessageHandlerAggregator.OnFlowProcessedAsync(state,flow, message, _applicationNode, linkedCts.Token);
+                        await _inputMessageHandlerAggregator.OnFlowProcessedAsync(state,flow, message, _applicationNode, linkedCts.Token);
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/FlowManager.cs:335]: duplicated line?

File: src/Take.Blip.Builder/FlowManager.cs [hunk 2]
```diff
@@ -331,7 +331,8 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
 
                         await ProcessGlobalOutputActionsAsync(context, flow, lazyInput, inputTrace, state, linkedCts.Token);
 
-                        await _inputMessageHandlerAggregator.OnFlowProcessedAsync(state, message, _applicationNode, linkedCts.Token);
+                        await _inputMessageHandlerAggregator.OnFlowProcessedAsync(state,flow, message, _applicationNode, linkedCts.Token);
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/FlowManager.cs:334]: put blank space after ","

File: src/Take.Blip.Builder/InputExpirationHandler.cs [hunk 3]
```diff
@@ -106,9 +108,15 @@ public async Task OnFlowProcessedAsync(State state, Message message, Node from,
                     throw new ArgumentException("Message content 'StateId' must be present", nameof(InputExpiration.StateId));
                 }
 
+                if (string.IsNullOrWhiteSpace(inputExpiration?.CurrentSessionState))
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/InputExpirationHandler.cs:111]: When the version is published, erroes will be raised due to previous jobs...
I think it's better not to throw an error when CurrentSessionState is null

File: src/Take.Blip.Builder/InputExpirationHandler.cs [hunk 4]
```diff
@@ -134,18 +142,23 @@ private bool IsMessageFromExpiration(Message message)
         /// </summary>
         /// <param name="state"></param>
         /// <param name="message"></param>
+        /// <param name="flow"></param>
         /// <returns></returns>
-        public bool IsValidateState(State state, Message message)
+        public bool IsValidateState(State state, Message message, Flow flow)
         {
             string stateToGo = string.Empty;
+            string currentSeessionState = string.Empty;
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/InputExpirationHandler.cs:150]: fix variable name...

File: src/Take.Blip.Builder/InputExpirationHandler.cs [hunk 5]
```diff
@@ -134,18 +142,23 @@ private bool IsMessageFromExpiration(Message message)
...
 
             return message?.Metadata?.TryGetValue(STATE_ID, out stateToGo) != true ||
                             string.IsNullOrEmpty(stateToGo) ||
                             stateToGo == IS_INPUT_EXPIRATION_FROM_SUBFLOW_REDIRECT ||
-                            state?.Id == stateToGo;
+                            (state?.Id == stateToGo) && message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState) != true ||
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/InputExpirationHandler.cs:155]: replace "message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState) != true" to "!message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState)"

Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/InputExpirationHandler.cs:155]: group "(state?.Id == stateToGo) && !message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState)" in parentheses...

parentheses ((state?.Id == stateToGo) && !message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState))

File: src/Take.Blip.Builder/InputMessageHandlerAggregator.cs [hunk 6]
```diff
@@ -52,11 +52,11 @@ public async Task OnFlowPreProcessingAsync(State state, Message message, Node fr
             }
         }
 
-        public async Task OnFlowProcessedAsync(State state, Message message, Node from, CancellationToken cancellationToken)
+        public async Task OnFlowProcessedAsync(State state, Flow flow, Message message, Node from, CancellationToken cancellationToken)
         {
             foreach (var handler in _handlers)
             {
-                await handler.OnFlowProcessedAsync(state, message, from, cancellationToken);
+                await handler.OnFlowProcessedAsync(state,flow, message, from, cancellationToken);
```
Comment (rafael-santiago-take) [inline src/Take.Blip.Builder/InputMessageHandlerAggregator.cs:59]: put blank space after ","

**Review by rafael-santiago-take — APPROVED** [review]

**Review by DanielFonsecaTake — APPROVED** [review]

**Pull request merged**